### ISSUE TITLE
Angular component - separate bindings to inputs and outputs

### ISSUE THREAD

**Issue by radek-anuszewski** [angular/angular.js#15913]
**I'm submitting a ...**

- [ ] bug report
- [x] feature request
- [ ] other (Please do not submit support requests here (see above))

**Current behavior:**

Today, with 1.5 version and greaters, I can declare `bindings `as:
```
bindings: {
  value: "<",
  name: "@",
  onChange: "&",
},
```

**Expected / new behavior:**

To be closer to Angular 2+, and to make code more intuitive with new trends about inputs and outputs, it can be separated to `inputs `and `outputs`: 
```
inputs: {
  value: "<",
  name: "@",
},
outputs: {
  onChange: "&",
},
```

**Angular version:** 1.6.4

**Browser:** [all]

**Anything else:**

It may be even join internally by `component `to `bindings`, so it won't be a breaking change I think, but for Angular's users it will make a difference with more separation and clarification.
Thank you in advance for considering this feature!

**Comment by gkalpak** [comment]
This is still different than Angular, so I don't see much value. It might be interesting to support `inputs`/`outputs` as arrays, which is what Angular does.

E.g.

```js
{
  ...
  inputs: ['value:<', 'name:@'],
  outputs: ['onChange:&']
}
```

...which is equivalent to Angular's...

```js
{
  ...
  inputs: ['value', 'name'],
  outputs: ['onChange']
}
```

...and AngularJS'...

```js
{
  ...
  bindings: {
    value: '<',
    name: '@',
    onChange: '&'
  }
}
```

**Comment by Narretz** [comment]
I agree with @gkalpak that this is not very useful, especially since it doesn't map 1:1 to Angular.

**Comment by radek-anuszewski** [comment]
@gkalpak solutions is also 100% ok for me.

**Comment by gkalpak** [comment]
I'll put it in the backlog. PRs are welcome (I guess)...
